### DISCUSSION TITLE
Will read from master if has replicas set up?

### DISCUSSION THREAD

**Discussion by kzzhao** [mikro-orm/mikro-orm#4818] [Q&A] [answered]
If `replicas` is specified, and `preferReadReplicas` is true, will read operations hit the master host?

const orm = await MikroORM.init({
  entities: [Author, ...],
  dbName: `my_database`,
  type: 'mysql',
  user: 'master_user',
  host: 'master_host',
  preferReadReplicas: true, // optional property, defaults to true
  replicas: [
    { name: 'read-1', host: 'read_host_1', user: 'read_user' },
    { name: 'read-2', host: 'read_host_2' }, // user omitted, will be taken from master connection
  ],
});

https://mikro-orm.io/docs/read-connections

**Comment by B4nan** [comment, marked as answer]
Nope, only the replicas.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/drivers/DatabaseDriver.ts#L101-L109

**Reply by kzzhao** [reply]
Thanks Martin! If I want to also read from master, I can add 'master_host' in the `replicas` right?

**Reply by B4nan** [reply]
Yes, that should do it.
